_posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc

```diff
@@ -0,0 +1,198 @@
+---
+layout: post
+title:  "Debug a provisioned WildFly server with the help of maven-wildfly-plugin"
+date:   2024-10-02
+tags:   wildfly maven-wildfly-plugin jberet
+author: liweinan
+description: Introducing the usage of maven-wildfly-plugin to debug a provisioned WildFly server.
+---
+
+== Debug a provisioned WildFly server with the help of maven-wildfly-plugin
+
+Sometimes I need to debug my project in a customized WildFly server. The requirements are:
+
+* I need to use a specific version of WildFly server.
+* I need to override some modules inside the WildFly server.
+* I need to run the WildFly server in debug mode, so I can use IDE tools to remotely debug the code.
+
+With the usage of `maven-wildfly-plugin`, the above goals can be achieved in an automated way. Here is an example of the configuration to show the usage of the plugin to achieve the above goals:
```

Comment:
s/usage of/use of the/g  (first time)

for the 2nd one either

s/usage/use

or

s/the usage of/how to use/g